EFCore.BulkExtensions.Tests/EFCoreBulkTestAsync.cs

```diff
@@ -221,7 +222,7 @@ private static async Task RunInsertOrUpdateAsync(bool isBulk, DbServerType dbSer
         }
         if (isBulk)
         {
-            var bulkConfig = new BulkConfig() { SetOutputIdentity = true, CalculateStats = true };
+            var bulkConfig = new BulkConfig() { SetOutputIdentity = true, CalculateStats = true, SqlBulkCopyOptions = SqlBulkCopyOptions.FireTriggers | SqlBulkCopyOptions.KeepIdentity};
```

Comment:
Now you got me confused, because the tests specifically set entity ids everywhere, how are they not dependent on locally set ids then?

Reply:
@videokojot I added a test case that reproduces the problem. My fix aligns the behavior with a non-MIT version of this library that we are trying to migrate from.

The problem was that the SQLite adapter attempted to insert the "id" column even when the identity insert was off. This results in a unique constraint violation when there is already an entity with "id=0" in the database. Those tests, where I added `SqlBulkCopyOptions.KeepIdentity`, seem to be working because the entity `id` is set, and they were not designed to rely on database-generated IDs. 

SQL Server seemed to not be affected by this problem because it is using BulkCopy behind the scenes, which probably handles it by itself.